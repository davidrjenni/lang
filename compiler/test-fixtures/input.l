{
	assert ~(~(true));
	assert ~(~(false));
	assert 27 = 3 + 5*5 - 1;
	assert false = true | true;
	assert -1 = 0 - 1 => true;
	let x := 2 * 3;
	let y := x * 3;
	assert x = 6;
	let z := true & x = 6;
	assert z;
}
