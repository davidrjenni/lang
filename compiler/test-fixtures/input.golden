
.macro AssertViolated
    movq $___fmt_assert, %rdi
    movq $0, %rax
    call printf
    movq $1, %rdi
    movq $0, %rax
    call exit
.endm

	.section .text
	.global main
main:
	pushq %rbp
	movq %rsp, %rbp
	subq $8, %rsp
	movb $1, %al  # test-fixtures/input.l:2:12
	cmpb $1, %al  # test-fixtures/input.l:2:12
	setne %al  # test-fixtures/input.l:2:11
	movb %al, %al  # test-fixtures/input.l:2:10
	cmpb $1, %al  # test-fixtures/input.l:2:10
	setne %al  # test-fixtures/input.l:2:9
	movb %al, %al  # test-fixtures/input.l:2:9
	cmpb $1, %al  # test-fixtures/input.l:2:9
	je .L1  # test-fixtures/input.l:2:2
	AssertViolated  # test-fixtures/input.l:2:2
.L1:
	movb $0, %al  # test-fixtures/input.l:3:12
	cmpb $1, %al  # test-fixtures/input.l:3:12
	setne %al  # test-fixtures/input.l:3:11
	movb %al, %al  # test-fixtures/input.l:3:10
	cmpb $1, %al  # test-fixtures/input.l:3:10
	setne %al  # test-fixtures/input.l:3:9
	movb %al, %al  # test-fixtures/input.l:3:9
	cmpb $1, %al  # test-fixtures/input.l:3:9
	je .L2  # test-fixtures/input.l:3:2
	AssertViolated  # test-fixtures/input.l:3:2
.L2:

	movq $0, %rax
	movq %rbp, %rsp
	popq %rbp
	ret

	.section .data
___fmt_assert: .string "Assertion violated\n"
